CHANGELOG.md

```diff
@@ -7,7 +7,7 @@
 - Add `az_json_writer_append_json_text()` to support appending existing JSON with the JSON writer.
 - Add support for system properties for IoT Hub messages to `az_iot_common.h`.
 - Add new HTTP result named `AZ_ERROR_HTTP_END_OF_HEADERS` to designate the end of the headers iterated over by `az_http_response_get_next_header()`.
-- Add new IoT result named `AZ_ERROR_IOT_NO_MORE_PROPERTIES` to designate the end of the properties iterated over by `az_iot_message_properties_next()`.
+- Add new IoT result named `AZ_ERROR_IOT_END_OF_PROPERTIES` to designate the end of the properties iterated over by `az_iot_message_properties_next()`.
```

Comment:
If we are going fully explicit/clear, like `AZ_ERROR_HTTP_END_OF_RESPONSE_HEADERS`
```suggestion
- Add new IoT result named `AZ_ERROR_IOT_END_OF_MESSAGE_PROPERTIES` to designate the end of the properties iterated over by `az_iot_message_properties_next()`.
```

@JeffreyRichter ?

Reply:
In general, I think the more specific the more likely the customer can diagnose and fix a bug. 
The IoT team will have to address the "other kinds of properties" part of your question.

Reply:
hmm, I'm not sure yet.  Will discuss with Dane.  Thanks.